docs/application/dotnet/guides/machine-learning/pipeline.md

```diff
@@ -0,0 +1,180 @@
+
+    ```xml
+    <privileges>
+      <!--To access media storage-->
+      <privilege>http://tizen.org/privilege/mediastorage</privilege>
+
+      <!--To access, read, and write to the external storage-->
+      <privilege>http://tizen.org/privilege/externalstorage</privilege>
+    </privileges>
+    ```
+
+<a name="construct"></a>
+## Pipeline construction
+
+To construct a pipeline, you must have a pipeline description with the [GStreamer](https://gstreamer.freedesktop.org) and [NNStreamer](https://github.com/nnstreamer/nnstreamer) elements.
+For more information, see the GStreamer manual or the NNStreamer documentation for examples and grammar.
+
+1. If the model file is located in the resource directory of your own application, you need to get its absolute path:
+
+    ```C#
+    string ResourcePath = Tizen.Applications.Application.Current.DirectoryInfo.Resource;
+    string model_path = ResourcePath + "models/mobilenet_v1_1.0_224_quant.tflite";
+    ```
+
+2. You can construct a pipeline with the description string, including the neural network framework and a model file:
+
+    ```C#
+    /* Create Pipeline instance with the pipeline description. */
+    string description = "appsrc name=srcx ! " +
+                         "other/tensor,dimension=(string)3:224:224:1,type=(string)uint8,framerate=(fraction)0/1 ! " +
+                         "tensor_filter framework=tensorflow-lite model=" + model_path + " ! " +
+                         "tensor_sink name=sinkx";
+    Pipeline pipe = new Pipeline(description);
+    ```
+
+<a name="inputdata"></a>
+## Push input data
+
+`SourceNode` provides a method to push input data into the pipeline:
+
+```C#
+/* Get the source node with the name 'srcx'. */
+var src_node = pipe.GetSource("srcx");
+
+/* Start the pipeline. */
+pipe.Start();
+
+/* Prepare input tensor data, with data type UInt8 and dimension 3:224:224:1. */
+var in_info = new TensorsInfo();
+in_info.AddTensorInfo(TensorType.UInt8, new int[4] { 3, 224, 224, 1 });
+
+var in_data = in_info.GetTensorsData();
+
+/* Push input tensor data into the pipeline. */
+src_node.Input(in_data);
+```
+
+<a name="outputdata"></a>
+## Subscribe to output data
+
+You can use `SinkNode` to get the output tensor data from the pipeline:
+
+```C#
+/* Firstly, declare the event handler for the data received event. */
+void DataReceivedEvent(object sender, DataReceivedEventArgs args)
+{
+    /* Add your task with received data. */
+    TensorsData dataReceived = args.Data;
+}
+```
+
+```C#
+/* Get the sink node with the name 'sinkx'. */
+var sink_node = pipe.GetSink("sinkx");
+
+/* Subscribe the event to get the data from the sink node. */
+sink_node.DataReceived += DataReceivedEvent;
+
+/* Start the pipeline. */
+pipe.Start();
+```
+
+<a name="dataflow"></a>
+## Data flow control
+
+You need to stop the data stream or select the data flow with multiple stream paths for neural network models. Pipeline provides methods to handle the flow.
```

Comment:
Yes, usage of auxiliary verbs like "may" and "might" always brings in an ambiguity. It gives a chance of possibility. Hence, we try to avoid using those in technical documents. 
Therefore, I have asked the author that the sentence is not clear and to verify if my understanding is correct?

For more information, see https://github.com/Samsung/tizen-docs/pull/1079#discussion_r407927336

Reply:
@mijin85cho 
I agree with your comment, usage of "You need to..." can change the meaning slightly.

original : You may need to stop the data stream, or to select the data flow with multiple stream paths for neural network models. Pipeline provides methods to handle the flow.

The first part is ambiguous, therefore, as I have asked in my previous comment, I would like to ask @jaeyun-jung to explain the sentence so that I can rephrase it better.

Reply:
This section is about the data flow : consider If developer needs this kind of flow control with complex pipeline (multiple inputs, outputs)

When creating simple pipeline (no valve/selector for flow), unnecessary.

Reply:
@jaeyun-jung,
Thank you for explaining.
How about rephrasing as:
For neural network models, if you need to stop the data stream or select the data flow with multiple stream paths, the Pipeline class provides methods to handle the flow.

or

If you need to stop the data stream or select the data flow with multiple stream paths for neural network models, the Pipeline class provides methods to handle the flow.

Please verify and correct me if I am wrong.

Reply:
Updated (If you need to ..)
please see the changes.